### PR TITLE
Switch connect v2 to slinky v1

### PR DESCRIPTION
**Pull request by chenyaoy**
### Changelist
Switch back from connect v2 to slinky v1 but with the new features in v2 sans the rename

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Release Notes

- **New Features**
  - Updated context handling across various methods to utilize `sdk.Context` instead of `context.Context`.
  - Introduced new message types and updated existing ones to reflect the transition from `connect` to `slinky`.

- **Bug Fixes**
  - Enhanced error handling in validation methods for better clarity in error messages.

- **Documentation**
  - Comments and documentation updated to reflect changes in service names and configurations.

- **Chores**
  - Updated import paths across multiple files to align with the new `slinky` package structure.
  - Modified CI/CD workflow to include additional branch triggers for release management.

### CODE REVIEW

File: protocol/go.mod [hunk 1]
```diff
@@ -69,7 +69,7 @@ require (
 	github.com/pelletier/go-toml v1.9.5
 	github.com/rs/zerolog v1.33.0
 	github.com/shopspring/decimal v1.3.1
-	github.com/skip-mev/connect/v2 v2.2.1
+	github.com/skip-mev/slinky v1.0.14-0.20241212193714-ab2537f1a1f2
```
Comment (chenyaoy) [inline protocol/go.mod:72]: TODO: Upgrade to released version

File: protocol/app/msgs/normal_msgs.go [hunk 2]
```diff
@@ -267,19 +267,19 @@ var (
...
+	NormalMsgsSlinky = map[string]sdk.Msg{
+		// slinky marketmap messages
+		"/slinky.marketmap.v1.MsgCreateMarkets":                   &marketmapmoduletypes.MsgCreateMarkets{},
+		"/slinky.marketmap.v1.MsgCreateMarketsResponse":           nil,
+		"/slinky.marketmap.v1.MsgParams":                          &marketmapmoduletypes.MsgParams{},
+		"/slinky.marketmap.v1.MsgParamsResponse":                  nil,
+		"/slinky.marketmap.v1.MsgRemoveMarketAuthorities":         &marketmapmoduletypes.MsgRemoveMarketAuthorities{},
+		"/slinky.marketmap.v1.MsgRemoveMarkets":                   &marketmapmoduletypes.MsgRemoveMarkets{},
+		"/slinky.marketmap.v1.MsgRemoveMarketsResponse":           nil,
+		"/slinky.marketmap.v1.MsgRemoveMarketAuthoritiesResponse": nil,
```
Comment (vincentwschau) [inline protocol/app/msgs/normal_msgs.go:279]: nit: Ordering? RemoveMarketAuthoritiesResponse should come after RemoveMarketAuthorities?

File: protocol/x/prices/keeper/slinky_adapter.go [hunk 3]
```diff
@@ -121,12 +111,12 @@ func (k Keeper) GetNumCurrencyPairs(ctx context.Context) (uint64, error) {
...
-func (k Keeper) GetNumRemovedCurrencyPairs(_ context.Context) (uint64, error) {
+func (k Keeper) GetNumRemovedCurrencyPairs(_ sdk.Context) (uint64, error) {
 	return 0, nil
 }
 
 // GetAllCurrencyPairs is not used with the DefaultCurrencyPair strategy.
-// See https://github.com/skip-mev/connect/v2/blob/main/abci/strategies/currencypair/default.go
-func (k Keeper) GetAllCurrencyPairs(_ context.Context) []slinkytypes.CurrencyPair {
+// See https://github.com/skip-mev/slinky/blob/main/abci/strategies/currencypair/default.go
+func (k Keeper) GetAllCurrencyPairs(ctx sdk.Context) []slinkytypes.CurrencyPair {
```
Comment (vincentwschau) [inline protocol/x/prices/keeper/slinky_adapter.go:120]: nit: Keep underscore naming to show it's unused?

**Review by vincentwschau — APPROVED** [review]

File: protocol/x/delaymsg/keeper/delayed_message.go [hunk 4]
```diff
@@ -188,7 +188,7 @@ func validateSigners(signers [][]byte) error {
 		)
 	}
 	if !bytes.Equal(signers[0], types.ModuleAddress) {
-		return errorsmod.Wrap(
+		return errorsmod.Wrapf(
```
Comment (aljo242) [inline protocol/x/delaymsg/keeper/delayed_message.go:191]: looks like this should still be Wrap

File: protocol/x/prices/keeper/market_param_test.go [hunk 5]
```diff
@@ -228,7 +228,7 @@ func TestModifyMarketParam_Errors(t *testing.T) {
 			minExchanges:       uint32(1),
 			minPriceChangePpm:  uint32(50),
 			exchangeConfigJson: validExchangeConfigJson,
-			expectedErr: errorsmod.Wrap(
+			expectedErr: errorsmod.Wrapf(
```
Comment (aljo242) [inline protocol/x/prices/keeper/market_param_test.go:231]: ditto

File: protocol/x/vest/types/vest_entry.go [hunk 6]
```diff
@@ -7,11 +7,11 @@ import (
 
 func (entry VestEntry) Validate() error {
 	if entry.VesterAccount == "" {
-		return errorsmod.Wrap(ErrInvalidVesterAccount, "vester account cannot be empty")
+		return errorsmod.Wrapf(ErrInvalidVesterAccount, "vester account cannot be empty")
```
Comment (aljo242) [inline protocol/x/vest/types/vest_entry.go:10]: ditto

**Review by vincentwschau — APPROVED** [review]

File: .github/workflows/protocol-build-and-push.yml [hunk 7]
```diff
@@ -6,7 +6,7 @@ on:  # yamllint disable-line rule:truthy
       - main
       - 'release/protocol/v[0-9]+.[0-9]+.x'  # e.g. release/protocol/v0.1.x
       - 'release/protocol/v[0-9]+.x'  # e.g. release/protocol/v1.x
-
+      - 'release/protocol/v[0-9]+.x'  # e.g. release/protocol/v1.x
```
Comment (vincentwschau) [inline .github/workflows/protocol-build-and-push.yml:9]: is this repeated?

Comment (chenyaoy) [reply .github/workflows/protocol-build-and-push.yml:9]: shoot, thanks

**Review by vincentwschau — APPROVED** [review]

**Comment by chenyaoy** [comment]
https://github.com/Mergifyio backport release/protocol/v8.x

**Pull request merged**